### PR TITLE
Add support for <system_name>/custom_builds.sh, update cee-rhel6 builds (TRIL-212)

### PR DESCRIPTION
**Pull request by bartlettroscoe**
@fryeguy52 

## Description

Adds a new feature which is a `<system_name>/custom_builds.sh` file.  This is used to define custom compiler and MPI compiler names.  I used this for an updated 'cee-rhel6' env.  This fixes the settings for all of the 'cee-rhel6' builds and now matches the current set of builds being used by SPARC on CEE RHEL6 machines (but not GPUs yet).

I also added a new simple local Ctest -S CDash driver script `ctest-s-local-test-driver.sh` to make it easy to submit results to Experimental CDash.

## Motivation and Context

Needed to update the 'cee-rhel6' env for the updated SPARC Trilinos configurations and this includes detailed compiler version info and the MPI implementation in the compiler name. See https://software-sandbox.sandia.gov/jira/browse/TRIL-212.

## How Has This Been Tested?

I used the new script `ctest-s-local-test-driver.sh` from every supported machine and env to submit to Experimental CDash as shown [here](https://testing.sandia.gov/cdash-dev-view/index.php?project=Trilinos&date=2018-11-17&filtercount=2&showfilters=1&filtercombine=and&field1=groupname&compare1=61&value1=Experimental&field2=buildname&compare2=65&value2=Trilinos-atdm-).

### CODE REVIEW

**Comment by bartlettroscoe** [comment]
@fryeguy52, @jmgate, @ibaned, @mhoemmen, or @rppawlo, any of you guys around this weekend to review and approve this?  IF there are issues with the documentation, I can fix that with a follow-up PR.  For now, I want to get this merged so that we can fix the 'cee-rhel6' builds (that got broken when SPARC changed the envs).

This is ready to go.

**Comment by mhoemmen** [comment]
Thanks Ross! :D

File: cmake/ctest/drivers/atdm/smart-jenkins-driver.sh [hunk 1]
```diff
@@ -20,7 +20,7 @@ if [ "${WORKSPACE}" == ""  ] ; then
   exit 1
 fi
 
-export ATDM_CONFIG_JOB_NAME=$JOB_NAME
+export ATDM_CONFIG_BUILD_NAME=$JOB_NAME
```
[the remark this answers is not in the document]
Comment (bartlettroscoe) [reply cmake/ctest/drivers/atdm/smart-jenkins-driver.sh]: > When backward-incompatible changes like this are made, can you please build in a backward-compatible layer, have the ATDM apps adjust for the new interface, and then rip out the backward-incompatible layer?

@jmgate, did this PR break the SPARC usage of the ATDM Trilinos configuration?  I did not think that EMPIRE was using  the script `cmake/ctest/drivers/atdm/smart-jenkins-driver.sh`, just the files `cmake/std/atdm/load-dev.sh` and `cmake/std/atd/ATDMDevEnv.cmake`.  For the usage of those files, they should be backward compatible due to

* https://github.com/bartlettroscoe/Trilinos/blob/fe5fb4a376bdae0ba101ca045057ecb38a6db02b/cmake/std/atdm/load-env.sh#L49

Is this not the case?

[the remark this answers is not in the document]
Comment (jmgate) [reply cmake/ctest/drivers/atdm/smart-jenkins-driver.sh]: Sorry, @bartlettroscoe, I probably commented in the wrong place.  EMPIRE doesn't use that script, but we did use the `ATDM_CONFIG_JOB_NAME` environment variable.  We had to update our scripts to set the `ATDM_CONFIG_BUILD_NAME` environment variable instead.  The variable name switch is what I was referring to.

**Comment by bartlettroscoe** [comment]
> I probably commented in the wrong place. EMPIRE doesn't use that script, but we did use the ATDM_CONFIG_JOB_NAME environment variable. We had to update our scripts to set the ATDM_CONFIG_BUILD_NAME environment variable instead. The variable name switch is what I was referring to.

@jmgate, can you point to what you had to update so that I understand your usage of that ATDM Trilinos configuration scripts?  It must be more complex than just calling `source <trilinosDir>/cmake/std/atdm/load-env.sh <build-name>`.

**Comment by jmgate** [comment]
See BuildScripts commit b5b9071.

**Pull request merged**